clang-tools-extra/test/clang-tidy/checkers/bsl-copy-move-access-specifier.cpp

```diff
@@ -0,0 +1,68 @@
+// RUN: %check_clang_tidy %s bsl-copy-move-access-specifier %t
+
+// Abstract base class
+class A {
+public:
+  A() = default;
+  A(A const &) = default;
+  // CHECK-MESSAGES: :[[@LINE-1]]:3: warning: Copy and move constructors shall be declared protected or defined “=delete” in base class; otherwise declare non-base class as final [bsl-copy-move-access-specifier]
+  A(A &&) = default;
+  // CHECK-MESSAGES: :[[@LINE-1]]:3: warning: Copy and move constructors shall be declared protected or defined “=delete” in base class; otherwise declare non-base class as final [bsl-copy-move-access-specifier]
+  virtual ~A() = 0;
```

Comment:
In all cases, you are using a virtual destructor. You need to add some tests for non-virtual destructors (both public and protected)

Reply:
Also need these for both the failure case and the good case